### PR TITLE
[ENHANCEMENT] Adds the ability to run generate on multiple addons at once

### PR DESCRIPTION
**Pull request by swelham**
This is my initial stab at #3650. This isn't quite ready for merge yet (needs tests at least), but I wanted some feedback on this as there are a couple of areas that don't sit well with me.

Firstly I am making the assumption that when running ```ember g addon1``` you cannot specify any options for that command (like you can in ```ember g model foo name:string```).

Secondly I am not to sure on the way that I am detecting if the blueprint is an addon or not. Currently looking in ```Blueprint#defaultLookupPaths()``` and ```Project#localBlueprintLookupPath()``` to work this out.

Overall though, this is working and doesn't break existing tests, so it can't be all that bad :)

### CODE REVIEW

File: lib/tasks/generate-from-blueprint.js [hunk 1]
```diff
@@ -12,8 +12,52 @@ module.exports = Task.extend({
...
+      paths: this.project.blueprintLookupPaths(),
+      ignoreMissing: ignoreMissing
+    });
+  },
+
+  runBlueprints: function(blueprints) {
+    return this.runBlueprint(blueprints.shift())
+      .then(function() {
+        if (blueprints.length > 0) {
+          this.runBlueprints(blueprints);
```
Comment (stefanpenner) [inline lib/tasks/generate-from-blueprint.js]: this needs a return, or we sever the promise chain

File: lib/tasks/generate-from-blueprint.js [hunk 2]
```diff
@@ -12,8 +12,52 @@ module.exports = Task.extend({
...
+
+  lookupBlueprint: function(name, ignoreMissing) {
+    return Blueprint.lookup(name, {
+      paths: this.project.blueprintLookupPaths(),
+      ignoreMissing: ignoreMissing
+    });
+  },
+
+  runBlueprints: function(blueprints) {
+    return this.runBlueprint(blueprints.shift())
```
Comment (stefanpenner) [inline lib/tasks/generate-from-blueprint.js]: this should likely use the sequence helper: https://github.com/ember-cli/ember-cli/blob/master/lib/utilities/sequence.js#L1-L15

Comment (swelham) [inline lib/tasks/generate-from-blueprint.js]: Awesome, I totally missed that helper.

File: lib/tasks/generate-from-blueprint.js (hunk 1 above)
Comment (swelham) [inline lib/tasks/generate-from-blueprint.js]: Ah yes I see. I will update to switch to using the sequence helper which will kill off this code anyway.

**Comment by stefanpenner** [comment]
@swelham any time coming up to work on

- [ ] using the sequence helper for the promise sequence
- [ ] tests

**Comment by swelham** [comment]
@stefanpenner I am currently away at the moment, I should have time to finish this on my return (about a week or so).

In terms of tests, do you see this just being an acceptance test in [generation-test.js]h(ttps://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/generate-test.js)?

**Comment by stefanpenner** [comment]
this seems to have gone stale. Would consider it, if rebased, tests passing and comments addressed.

Closing for now.

**Pull request closed without merging**